From 742587dce258adf436f360eed183a7ca0c58fa18 Mon Sep 17 00:00:00 2001
From: "Jasper St. Pierre" <jstpierre@mecheye.net>
Date: Wed, 2 Jan 2013 09:05:32 -0500
Subject: [PATCH] main: Don't mess up the modal stack when the focus actor is
 destroyed

This seems to be an incorrect conversion when we moved from an array
to an object of keys in 3a6b4f3.

https://bugzilla.gnome.org/show_bug.cgi?id=690965
---
 js/ui/main.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/main.js b/js/ui/main.js
index 12daee2..1334d58 100644
--- a/js/ui/main.js
+++ b/js/ui/main.js
@@ -553,7 +553,7 @@ function pushModal(actor, params) {
         curFocusDestroyId = curFocus.connect('destroy', function() {
             let index = _findModal(actor);
             if (index >= 0)
-                modalActorFocusStack[index].actor = null;
+                modalActorFocusStack[index].focus = null;
         });
     }
     modalActorFocusStack.push({ actor: actor,
-- 
1.8.0.2